paymentapi/src/main/java/fi/hel/verkkokauppa/payment/service/OnlinePaymentService.java

```diff
@@ -36,23 +65,90 @@ public String createFromOrder(OrderPaymentDto dto) {
 
         // TODO use the common payment configuration values
 
-        // construct a Payment entity
-        Payment payment = createPayment(dto);
-        String paymentId = payment.getPaymentId();
+        try {
+            String token = tokenFetcher.getToken(payloadBuilder.buildFor(dto));
+
+            Payment payment = createPayment(dto, paymentType, isRecurringOrder ? token : null);
```

Comment:
Token should be set for payment no matter if its recurring or not. This token that is coming from vismapay is not the one for recurring transactions. So this token is payment identification token not the card_token.